### PR TITLE
fix(ui/acl): ACL config menu subgroups align with parent

### PR DESCRIPTION
**Pull request by jdelpierre**
Unable to select subgroup access options.
Selected option (radio buttons) of parent groups should be correctly applied the appropriate matching options in the subgroup

Refs: MON-5341

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by sc979 — COMMENTED** [review]
Seems fine.
But kindly correct the indentation from 6 spaces to 2 (or at least to 4) spaces

**Comment by jdelpierre** [comment]
> Seems fine.
> But kindly correct the indentation from 6 spaces to 2 (or at least to 4) spaces

On just the part of modification ? 
Because it's a big commit about all the file content, that's why I left the consistency with the whole

File: www/include/common/javascript/tool.js [hunk 1]
```diff
@@ -32,171 +32,166 @@
...
+      if (element.type === 'checkbox') {
+        input.checked = element.checked;
+      }
+
+      if (element.type === 'radio') {
+        if (input.type === 'radio') {
+          input.checked = input.value === valueInputParent ? true : false;
+        }
+        if (input.type === 'checkbox') {
+          input.checked = [1, 2].includes(parseInt(valueInputParent)) ? true : false;
```
Comment (kduret) [inline www/include/common/javascript/tool.js:112]:
```suggestion
          input.checked = [1, 2].includes(parseInt(valueInputParent));
```

File: www/include/common/javascript/tool.js [hunk 2]
```diff
@@ -32,171 +32,166 @@
...
-			value = 2;
-		}
-		for (var j = 0; j < element.length; j++) {
-			if (element[j].value == value) {
-				element[j].checked = true;
-			}
-		}
-	}
+function checkItem(element, toCheck) {
+  if (element.type == 'checkbox') {
```
Comment (kduret) [inline www/include/common/javascript/tool.js:36]: I know it's just formatting but I think we can update this safely
```suggestion
  if (element.type === 'checkbox') {
```

File: www/include/common/javascript/tool.js [hunk 3]
```diff
@@ -32,171 +32,166 @@
...
+    if (toCheck) {
+      element.checked = true;
+    } else {
+      element.checked = false;
+    }
+  } else if (element.type == 'radio') {
```
Comment (kduret) [inline www/include/common/javascript/tool.js:42]: I know it's just formatting but I think we can update this safely
```suggestion
  } else if (element.type === 'radio') {
```

File: www/include/common/javascript/tool.js [hunk 4]
```diff
@@ -32,171 +32,166 @@
...
+    var element = document.Form[element.name];
+    var value = 0;
+    if (toCheck) {
+      value = 2;
+    }
```
Comment (kduret) [inline www/include/common/javascript/tool.js:47]:
```suggestion
    const value = toCheck ? 2 : 0;
    var value = 0;
    if (toCheck) {
      value = 2;
    }
```

File: www/include/common/javascript/tool.js [hunk 5]
```diff
@@ -32,171 +32,166 @@
...
-	var element = document.Form[element.name];
-	for (var j = 0; j < element.length; j++) {
-		if (element[j].checked) {
-			if (element[j].value > 0) {
-				return true;
-			}
-		}
-	}
-	return false;
+  if (element.type == 'checkbox') {
```
Comment (kduret) [inline www/include/common/javascript/tool.js:57]: I know it's just formatting but I think we can update this safely
```suggestion
  if (element.type === 'checkbox') {
```

File: www/include/common/javascript/tool.js [hunk 6]
```diff
@@ -32,171 +32,166 @@
...
+    Object.values(inputs).forEach((input) => {
+      let valueInputParent = element.value;
+
+      if (element.type === 'checkbox') {
+        input.checked = element.checked;
+      }
+
+      if (element.type === 'radio') {
+        if (input.type === 'radio') {
+          input.checked = input.value === valueInputParent ? true : false;
```
Comment (kduret) [inline www/include/common/javascript/tool.js:109]:
```suggestion
          input.checked = input.value === valueInputParent;
```

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**